web-client/src/services/rolebindingRequests.tsx

```diff
@@ -0,0 +1,203 @@
+import {templateClusterResourceRolePrefix} from "../constants";
+import {httpClient} from "./httpClient";
+import {ClusterAccess} from "../components/types";
+import {AxiosInstance, AxiosResponse} from "axios";
+import {AggregatedRoleBinding} from "./role";
+
+interface HasAddGeneratedUser {
+  readonly addGeneratedForUser: boolean
+  
+}
+
+interface CreateClusterRoleBindingParameters extends HasAddGeneratedUser {
+  readonly template: string,
+  readonly username: string,
+  readonly clusterRolebindingName: string
+}
+
+interface CreateClusterRolebindingParameters extends HasAddGeneratedUser {
+  readonly clusterAccess: ClusterAccess,
+  readonly username: string
+  
+}
+
+
+interface CreateRolebindingParameters extends HasAddGeneratedUser {
+  readonly template: string,
+  readonly username: string,
+  readonly namespace: string,
+  readonly roleBindingName: string
+}
+
+interface CreateAllRolebindingsParameters {
+  readonly aggregatedRoleBindings: AggregatedRoleBinding[],
+  readonly username: string,
+  readonly clusterAccess: ClusterAccess
+}
+
+/**
+ * This class contains all the httpRequests for the clusterRoleBinding, roleBinding creation.
+ */
+class RolebindingCreateRequests {
+  
+  
+  constructor(private readonly httpClient: AxiosInstance) {
+  }
+  
+  private getTemplateName(clusterAccess: ClusterAccess): string {
+    switch (clusterAccess) {
+      case "none":
+        throw new Error("case 'none' not supported");
+      case "read":
+        return templateClusterResourceRolePrefix + 'read-only';
+      case "write":
+        return templateClusterResourceRolePrefix + 'admin';
+      default:
+        throw new Error(`unrecognized case: ${clusterAccess}`)
+      
+    }
+  }
+  
+  /**
+   * executes and encapsulates the common logic for the http requests towards the '/api/create-cluster-rolebinding' endpoint.
+   *
+   * @param params
+   */
+  private async httpCreateClusterRolebinding(params: CreateClusterRoleBindingParameters) {
```

Comment:
I wouldn't abuse the interfaces as a means to aggregate arguments to give to a function